File: win-prog/ProgWinVB/Lines Curves and Area Fills/XMarksTheSpot/XMarksTheSpot.vb
'----------------------------------------------
' XMarksTheSpot.vb (c) 2002 by Charles Petzold
'----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class XMarksTheSpot
    Inherits Form

    Shared Sub Main()
        Application.Run(New XMarksTheSpot())
    End Sub

    Sub New()
        Text = "X Marks The Spot"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText
        ResizeRedraw = True
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim pn As New Pen(ForeColor)

        grfx.DrawLine(pn, 0, 0, _
                      ClientSize.Width - 1, ClientSize.Height - 1)
        grfx.DrawLine(pn, 0, ClientSize.Height - 1, _
                      ClientSize.Width - 1, 0)
    End Sub
End Class

File: win-prog/ProgWinCS/Lines Curves and Area Fills/XMarksTheSpot/XMarksTheSpot.cs
//--------------------------------------------
// XMarksTheSpot.cs © 2001 by Charles Petzold
//--------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class XMarksTheSpot: Form
{
     public static void Main()
     {
          Application.Run(new XMarksTheSpot());
     }
     public XMarksTheSpot()
     {
          Text = "X Marks The Spot";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;
          ResizeRedraw = true;
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics grfx = pea.Graphics;
          Pen      pen  = new Pen(ForeColor);

          grfx.DrawLine(pen, 0, 0, 
                             ClientSize.Width - 1, ClientSize.Height - 1);
          grfx.DrawLine(pen, 0, ClientSize.Height - 1, 
                             ClientSize.Width - 1, 0);
     }
}
